# Simulation settings for using moveit_sim_controllers
moveit_sim_hw_interface:
  joint_model_group: moving_base
  joint_model_group_pose: very_end1
# Settings for ros_control_boilerplate control loop
generic_hw_control_loop:
  loop_hz: 300
  cycle_time_error_threshold: 0.01
# Settings for ros_control hardware interface
hardware_interface:
  joints:
    - world_joint
    - shoulder_pan_joint
    - shoulder_lift_joint
    - elbow_joint
    - wrist_1_joint
    - wrist_2_joint
    - wrist_3_joint
    - gripper_joint
    - gripper_joint1
    - gripper_joint2
    - gripper_joint3
    - gripper_joint4
    - gripper_joint5
    - gripper_joint6
    - gripper_joint7
    - gripper_joint8
  sim_control_mode: 1  # 0: position, 1: velocity
# Publish all joint states
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
controller_list:
  - name: moving_base_controller
    action_ns: follow_joint_trajectory
    default: True
    type: FollowJointTrajectory
    joints:
      - world_joint
    gains: # Required because we're controlling a velocity interface
      world_joint: {p: o,  d: 1, i: 1, i_clamp: 1} # Smaller 'p' term, since ff term does most of the 
  - name: ur5_arm_controller
    action_ns: follow_joint_trajectory
    default: True
    type: FollowJointTrajectory
    joints:
      - shoulder_pan_joint
      - shoulder_lift_joint
      - elbow_joint
      - wrist_1_joint
      - wrist_2_joint
      - wrist_3_joint

moving_base_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - world_joint
  gains:
    world_joint:
      p: 7000
      d: 50
      i: 10
      i_clamp: 100
ur5_arm_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - shoulder_pan_joint
    - shoulder_lift_joint
    - elbow_joint  # Modified joint name
    - wrist_1_joint
    - wrist_2_joint
    - wrist_3_joint
  gains:
    shoulder_pan_joint:
      p: 7000
      d: 50
      i: 10
      i_clamp: 100
    shoulder_lift_joint:
      p: 7000
      d: 30
      i: 100
      i_clamp: 400
    elbow_joint:
      p: 7000
      d: 50
      i: 10
      i_clamp: 100
    wrist_1_joint:
      p: 300
      d: 20
      i: 10
      i_clamp: 100
    wrist_2_joint:
      p: 200
      d: 10
      i: 0.1
      i_clamp: 100
    wrist_3_joint:
      p: 100
      d: 1
      i: 0.1
      i_clamp: 100
